docs/tasks/stateful-sets/deleting-pods.md

```diff
@@ -0,0 +1,70 @@
+
+### Deletion of Pods
+
+#### Graceful Deletion
+
+Graceful deletion may be performed using the following:
+
+```shell
+kubectl delete pods <pod>
+```
+
+For the above to lead to graceful termination, the pod **must not** specify a `pod.Spec.TerminationGracePeriodSeconds` of 0. The practice of setting a `pod.Spec.TerminationGracePeriodSeconds` of 0 seconds is unsafe and strongly discouraged for Stateful Set Pods. Graceful deletion is safe and will ensure that the [Pod shuts down gracefully](/docs/user-guide/pods/#termination-of-pods) before the kubelet deletes the name from the apiserver. 
+
+Kubernetes (versions 1.5 or newer) will not delete Pods just because a Node is unreachable. The Pods running on an unreachable Node enter the 'Terminating' or 'Unknown' state eventually. Pods may also enter these states when the user attempts graceful deletion (*link to kubectl documentation*) of a pod on an unreachable Node. The only ways in which a pod in such a state can be removed from the apiserver are as follows:
+   * The Node object is deleted (either by the user, or by the node controller).
+   * The kubelet on the unresponsive Node starts responding, kills the Pod and removes the entry from the apiserver. 
+   * Force deletion of the pod by the user.
+   
+It is preferred to use the first or second approaches. If a Node is confirmed to be dead (e.g. permanently disconnected from the network, powered down, etc) then the node object should be deleted. If the node is suffering from a network partition, then try to resolve this or wait for it to resolve. When the partition heals, the kubelet will complete the deletion of the pod and free up its name in the apiserver. 
+
+Normally, the system completes the deletion once the Pod is no longer running on a Node, or the Node is deleted by an administrator. You may override this by force deleting the pod.
+
+#### Force Deletion
+
+Force deletions do **not** wait for confirmation from the kubelet that the pod has been terminated. Irrespective of whether a force deletion is successful in killing a pod, it will immediately free up the name from the apiserver. This would let the Stateful Set controller create a replacement pod with that same identity. However, the risks of doing so must be evident. If there is a pod running on a Node that has been partitioned from the apiserver, it may still be able to communicate with other Pods in the Stateful Set. Force deletion of a Stateful Set Pod is a user's assertion that the Pod in question will never again make contact with other Pods in the Set and its name can be safely freed up for a replacement to be created. 
```

Comment:
I'm against "must be evident." We need to spell these things out.

"This would allow the StatefulSet controller to create a replacement pod with the same identity; this can lead to the duplication of a still-running pod, and if said pod can still communicate with the other members of the StatefulSet, will violate the *at most one* semantics that StatefulSet is designed to guarantee."